dtfs-central-api/src/repositories/tfm-users-repo/tfm-users.repo.ts

```diff
@@ -0,0 +1,23 @@
+import { Collection, WithoutId, ObjectId } from 'mongodb';
+import { MONGO_DB_COLLECTIONS, TfmUser } from '@ukef/dtfs2-common';
+import { mongoDbClient } from '../../drivers/db-client';
+
+export class TfmUsersRepo {
+  /**
+   * Gets the tfm users collection
+   * @returns The collection
+   */
+  private static async getCollection(): Promise<Collection<WithoutId<TfmUser>>> {
+    return mongoDbClient.getCollection(MONGO_DB_COLLECTIONS.TFM_USERS);
+  }
+
+  /**
+   * Finds a user by id
+   * @param id - The user id
+   * @returns The found user
+   */
+  public static async findOneUserById(id: string | ObjectId): Promise<TfmUser | null> {
```

Comment:
try-catch please?

Reply:
The responsibility of handling any errors thrown by this function belongs to the caller, so this function itself should not have a `try-catch`

Reply:
I would prefer `try-catch` so a calling function which has not implemented is not crashed and also reduced overall number of lines across the functions which do not necessarily need to implement any exceptional handling.

Reply:
I'm not sure I agree with that logic.

Firstly, any function which calls this method is likely already wrapped in a try-catch and, if it isn't, I'd argue that means the function is poorly designed as you should not be querying a db without proper error handling (in fact, you shouldn't really have any async code which occurs without proper error handling)

The second thing is a semantic issue with this function - returning `null` implies that a user was not found, meaning the caller of this function is likely to interpret this as a `404` and respond in turn. However, if the `null` return came because an error was thrown rather than a user not being found, this would make debugging very difficult as the logging would say "404: User not found". This would cause the developer to start trying to figure out why the user does not exist rather than seeing that something went wrong with the actual db operation.

Reply:
I think a try/catch here can only help. IMO, any async call should have a try/catch handler.

However it looks like all or the majority of other functions in the `repositories` directory do _not_ have a try/catch, therefore I think it should remain as is (without try/catch), otherwise they'll have different patterns.

Reply:
Resolving this so that we follow the same pattern and we can get this PR merged